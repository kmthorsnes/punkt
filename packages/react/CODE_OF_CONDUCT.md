# Retningslinjer for å bidra

Her skriver vi vår code of conduct.