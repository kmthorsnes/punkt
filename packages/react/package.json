{
  "name": "@oslokommune/punkt-react",
  "version": "11.4.2",
  "description": "React komponentbibliotek til Punkt, et designsystem laget av Oslo Origo",
  "homepage": "https://punkt.oslo.kommune.no",
  "author": "Team Designsystem, Oslo Origo",
  "main": "dist/punkt-react.umd.js",
  "module": "dist/punkt-react.es.js",
  "files": [
    "/dist",
    "src/components/**/*",
    "CHANGELOG.md",
    "CONTRIBUTING.md"
  ],
  "type": "module",
  "types": "dist/index.d.ts",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "import": "./dist/punkt-react.es.js",
      "require": "./dist/punkt-react.umd.js"
    }
  },
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "build-app": "tsc && vite build --config vite.config-app.ts",
    "lint": "eslint 'src/**/*.{js,jsx,ts,tsx}'",
    "lint:fix": "eslint --fix 'src/**/*.{jsx,ts,tsx}'",
    "format": "prettier --write src//**/*.{ts,tsx,css} --config ./.prettierrc",
    "preview": "vite preview --outDir dist-app",
    "test": "react-scripts test --env=jsdom"
  },
  "devDependencies": {
    "@oslokommune/punkt-assets": "^11.0.0",
    "@oslokommune/punkt-css": "^11.4.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.3",
    "@types/jest-axe": "^3.5.5",
    "@types/node": "^18.16.19",
    "@types/react": "^18.2.14",
    "@types/react-dom": "^18.2.6",
    "@vitejs/plugin-react": "^3.1.0",
    "classnames": "^2.3.2",
    "eslint": "^8.56.0",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-prettier": "^5.1.0",
    "eslint-plugin-simple-import-sort": "^10.0.0",
    "jest": "^29.6.1",
    "jest-axe": "^8.0.0",
    "jest-environment-jsdom": "^29.6.1",
    "prettier": "^3.1.1",
    "react-hooks": "^1.0.1",
    "react-router-dom": "^6.14.1",
    "react-scripts": "^5.0.1",
    "sass": "^1.63.6",
    "typescript": "^4.9.3",
    "vite": "^4.4.12",
    "vite-plugin-dts": "^3.6.4"
  },
  "peerDependencies": {
    "@oslokommune/punkt-assets": ">=3.0.0",
    "@oslokommune/punkt-css": ">=3.0.0",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  },
  "engines": {
    "node": ">=16.0.0"
  },
  "private": false,
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "keywords": [
    "punkt",
    "designsystem",
    "react",
    "reactjs",
    "components",
    "ux",
    "ui"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/oslokommune/punkt.git"
  },
  "bugs": {
    "url": "https://github.com/oslokommune/punkt/issues"
  },
  "license": "MIT"
}
