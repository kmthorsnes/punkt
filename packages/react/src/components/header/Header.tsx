import classNames from 'classnames'
import React, { ForwardedRef, forwardRef, HTMLAttributes } from 'react'

import { PktButton } from '../button/Button'
import { PktIcon } from '../icon/Icon'

export interface User {
  name?: string
  shortname?: string
  lastLoggedIn?: string | Date
}

export interface UserMenuItem {
  iconName?: string
  title: string
  target: string | (() => void)
}

export interface Representing {
  name?: string
  shortname?: string
  orgNumber?: string | number
}

export interface UserMenuFooterItem {
  title: string
  target: string | (() => void)
}

export interface IPktHeader extends HTMLAttributes<HTMLDivElement> {
  logoLink?: string | (() => void)
  serviceName?: string
  fixed?: boolean
  scrollToHide?: boolean
  user?: User
  userMenu?: UserMenuItem[]
  representing?: Representing
  userOptions?: UserMenuItem[]
  userMenuFooter?: UserMenuFooterItem[]
  canChangeRepresentation?: boolean
  showMenuButton?: boolean
  showLogOutButton?: boolean
  openMenu?: () => void
  logOut?: () => void
  changeRepresentation?: () => void
}

export const PktHeader = forwardRef(
  (
    {
      logoLink = 'https://www.oslo.kommune.no/',
      serviceName,
      fixed = true,
      scrollToHide = true,
      user,
      userMenu,
      representing,
      userOptions,
      userMenuFooter,
      canChangeRepresentation = true,
      showMenuButton = false,
      showLogOutButton = false,
      openMenu,
      logOut,
      changeRepresentation,
      ...props
    }: IPktHeader,
    ref: ForwardedRef<HTMLDivElement>,
  ) => {
    const lastLoggedIn = React.useMemo(() => {
      if (typeof user?.lastLoggedIn === 'string') {
        return user.lastLoggedIn
      }
      return user?.lastLoggedIn
        ? new Date(user.lastLoggedIn).toLocaleString('nb-NO', {
            year: 'numeric',
            month: 'long',
            day: 'numeric',
          })
        : ''
    }, [user])

    const [hidden, setHidden] = React.useState(false)
    const [lastScrollPosition, setLastScrollPosition] = React.useState(0)
    const [userMenuOpen, setUserMenuOpen] = React.useState(false)

    const userMenuRef = React.useRef<HTMLLIElement>(null)

    React.useEffect(() => {
      document.addEventListener('mouseup', clickOutside)
      window.addEventListener('scroll', onScroll)
      return () => {
        document.removeEventListener('mouseup', clickOutside)
        window.removeEventListener('scroll', onScroll)
      }
    })

    const openUserMenu = () => {
      setUserMenuOpen(!userMenuOpen)
    }

    const clickOutside = (e: MouseEvent) => {
      if (userMenuRef.current && !userMenuRef.current.contains(e.target as HTMLElement)) {
        setUserMenuOpen(false)
      }
    }

    const onScroll = () => {
      if (scrollToHide) {
        const currentScrollPosition = window.pageYOffset || document.documentElement.scrollTop
        if (currentScrollPosition < 0) {
          return
        }
        if (Math.abs(currentScrollPosition - lastScrollPosition) < 60) {
          return
        }
        setHidden(currentScrollPosition > lastScrollPosition)
        setLastScrollPosition(currentScrollPosition)
      }
    }

    return (
      <header
        {...props}
        id="pkt-header"
        data-testid="pkt-header"
        aria-label="Topp"
        className={classNames('pkt-header', {
          'pkt-header--fixed': fixed,
          'pkt-header--scroll-to-hide': scrollToHide,
          'pkt-header--hidden': hidden,
        })}
        ref={ref}
      >
        <div className="pkt-header__logo">
          {typeof logoLink === 'string' ? (
            <a aria-label="Tilbake til forside" className="pkt-header__logo-link" href={logoLink}>
              <PktIcon
                name="oslologo"
                className="pkt-header__logo-svg"
                aria-hidden="true"
                path="https://punkt-cdn.oslo.kommune.no/latest/logos/"
              ></PktIcon>
            </a>
          ) : (
            <button
              aria-label="Tilbake til forside"
              className="pkt-link-button pkt-link pkt-header__logo-link"
              onClick={logoLink}
            >
              <PktIcon
                name="oslologo"
                className="pkt-header__logo-svg"
                aria-hidden="true"
                path="https://punkt-cdn.oslo.kommune.no/latest/logos/"
              ></PktIcon>
            </button>
          )}
          <span className="pkt-header__logo-service">{serviceName}</span>
        </div>
        <nav className="pkt-header__actions">
          <ul className="pkt-header__actions-row">
            {showMenuButton && (
              <li>
                <PktButton
                  className="pkt-header__menu-btn"
                  skin="secondary"
                  variant="icon-right"
                  iconName="menu"
                  onClick={openMenu}
                >
                  Meny
                </PktButton>
              </li>
            )}
            {(user || representing) && (
              <li
                data-testid="usermenu"
                className={`pkt-header--has-dropdown ${
                  userMenuOpen && !hidden ? 'pkt-header--open-dropdown' : ''
                }`}
                ref={userMenuRef}
              >
                <button
                  className="pkt-header__user-btn pkt-btn pkt-btn--secondary pkt-btn--icons-right-and-left"
                  type="button"
                  role="button"
                  aria-controls="pktUserDropdown"
                  aria-expanded={userMenuOpen}
                  onClick={openUserMenu}
                >
                  <PktIcon name="user" className="pkt-btn__icon" />
                  <span className="pkt-header__user-fullname">
                    {representing?.name || user?.name}
                  </span>
                  <span className="pkt-header__user-shortname">
                    {representing?.shortname || user?.shortname}
                  </span>
                  <PktIcon name="chevron-thin-down" className="pkt-btn--closed" />
                  <PktIcon name="chevron-thin-up" className="pkt-btn--open" />
                </button>
                <ul id="pktUserDropdown" className="pkt-header__dropdown pkt-user-menu">
                  {user && (
                    <li>
                      <div className="pkt-user-menu__label">Pålogget som</div>
                      <div className="pkt-user-menu__name">{user.name}</div>
                      {user.lastLoggedIn && (
                        <div className="pkt-user-menu__last-logged-in">
                          Sist pålogget: <time>{lastLoggedIn}</time>
                        </div>
                      )}
                    </li>
                  )}
                  {userMenu && (
                    <li>
                      <ul className="pkt-list">
                        {userMenu.map((item, index) => (
                          <li key={`userMenu-${index}`}>
                            {typeof item.target === 'string' ? (
                              <a href={item.target} className="pkt-link">
                                {item.iconName && (
                                  <PktIcon name={item.iconName} className="pkt-link__icon" />
                                )}
                                {item.title}
                              </a>
                            ) : (
                              <button className="pkt-link-button pkt-link" onClick={item.target}>
                                {item.iconName && (
                                  <PktIcon name={item.iconName} className="pkt-link__icon" />
                                )}
                                {item.title}
                              </button>
                            )}
                          </li>
                        ))}
                      </ul>
                    </li>
                  )}
                  {(representing || canChangeRepresentation) && (
                    <li>
                      {representing && (
                        <>
                          <div className="pkt-user-menu__label">Representerer</div>
                          <div className="pkt-user-menu__name">{representing.name}</div>
                          {representing.orgNumber && (
                            <div className="pkt-user-menu__org-number">
                              Org.nr. {representing.orgNumber}
                            </div>
                          )}
                        </>
                      )}
                      <ul className="pkt-list mt-size-16">
                        {canChangeRepresentation && (
                          <li>
                            <button
                              className="pkt-link-button pkt-link"
                              onClick={changeRepresentation}
                            >
                              <PktIcon name="cogwheel" className="pkt-link__icon" />
                              Endre organisasjon
                            </button>
                          </li>
                        )}
                      </ul>
                    </li>
                  )}
                  <li>
                    <ul className="pkt-list">
                      {(userOptions || !showLogOutButton) && (
                        <>
                          {userOptions?.map((item, index) => (
                            <li key={`userOptions-${index}`}>
                              {typeof item.target === 'string' ? (
                                <a href={item.target} className="pkt-link">
                                  {item.iconName && (
                                    <PktIcon name={item.iconName} className="pkt-link__icon" />
                                  )}
                                  {item.title}
                                </a>
                              ) : (
                                <button className="pkt-link-button pkt-link" onClick={item.target}>
                                  {item.iconName && (
                                    <PktIcon name={item.iconName} className="pkt-link__icon" />
                                  )}
                                  {item.title}
                                </button>
                              )}
                            </li>
                          ))}
                          {!showLogOutButton && (
                            <li>
                              <button className="pkt-link-button pkt-link" onClick={logOut}>
                                <PktIcon name="exit" className="pkt-link__icon" />
                                Logg ut
                              </button>
                            </li>
                          )}
                        </>
                      )}
                    </ul>
                  </li>
                  {userMenuFooter && (
                    <li className="footer">
                      <ul className="pkt-list-horizontal bordered">
                        {userMenuFooter.map((item, index) => (
                          <li key={`userMenuFooter-${index}`}>
                            {typeof item.target === 'string' ? (
                              <a href={item.target} className="pkt-link">
                                {item.title}
                              </a>
                            ) : (
                              <button className="pkt-link-button pkt-link" onClick={item.target}>
                                {item.title}
                              </button>
                            )}
                          </li>
                        ))}
                      </ul>
                    </li>
                  )}
                </ul>
              </li>
            )}
            {showLogOutButton && (
              <li>
                <PktButton
                  className="pkt-header__user-btn pkt-header__user-btn-logout"
                  iconName="exit"
                  role="button"
                  onClick={logOut}
                  skin="secondary"
                  variant="icon-right"
                >
                  Logg ut
                </PktButton>
              </li>
            )}
          </ul>
        </nav>
      </header>
    )
  },
)
