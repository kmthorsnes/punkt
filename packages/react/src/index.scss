@use 'sass:map';
@use 'pkt/scss/abstracts/variables' with ($font-path: 'pktAssets'
);
@use 'pkt/scss/abstracts/functions';
@use 'pkt/scss/abstracts/mixins/breakpoints' as *;
@use 'pkt/scss/abstracts/mixins/typography';
@use 'pkt/scss/pkt.scss';

body {
  background-color: #efefef;
  margin: 0;
  min-width: 360px;
  padding: 0;
}

h1 {
  @include typography.get-text('pkt-txt-28');
  margin-bottom: map.get(variables.$spacing, 'size-20');

  @include bp('tablet-up') {
    @include typography.get-text('pkt-txt-36');
  }
}

h2 {
  @include typography.get-text('pkt-txt-24');
  margin-bottom: map.get(variables.$spacing, 'size-10');

  @include bp('tablet-up') {
    @include typography.get-text('pkt-txt-30');
  }
}

p {
  margin-bottom: map.get(variables.$spacing, 'size-15');
}

.pkt-logo {
  width: 50%;
  height: 50%;
  max-width: 18rem;
  display: inline-block;
  margin: 1rem;
}

.page-container {
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 1fr;
  grid-column-gap: 0;
  grid-row-gap: 0;
  grid-template-areas:
    'header'
    'main';
}

.dev-header {
  $-header-height: 50px;
  $-header-height-phablet-up: 60px;
  $-logo-width: calc($-header-height * 60 / 31);
  $-logo-width-phablet-up: calc($-header-height-phablet-up * 60 / 31);

  background-color: var(--pkt-color-background-default);
  border-bottom: 1px solid var(--pkt-color-border-subtle);
  display: grid;
  gap: 0 0;
  grid-area: header;
  grid-template-areas: 'dev-header-logo . dev-header-nav';
  grid-template-columns: minmax($-logo-width, $-logo-width-phablet-up) 1fr auto auto;
  padding: map.get(variables.$spacing, 'size-20') map.get(variables.$spacing, 'size-15');
  align-items: center;

  &__logo {
    grid-area: dev-header-logo;

    &-link {
      display: block;
    }

    &-svg {
      height: $-header-height;
      width: calc($-header-height * 60 / 31);

      @include bp('phablet-up') {
        height: $-header-height-phablet-up;
        width: calc($-header-height-phablet-up * 60 / 31);
      }
    }
  }

  &__nav {
    grid-area: dev-header-nav;
  }
}

.component {
  background-color: #fff;
  margin-bottom: map.get(variables.$spacing, 'size-40');
  padding: 10px;
  transition: all 0.3s ease-out;

  [data-mode='dark'] & {
    background-color: var(--pkt-color-brand-dark-blue-1000);
    color: #fff;
  }

  &__example {
    align-items: flex-start;
    display: flex;
    flex-wrap: wrap;
    gap: 5px 5px;
  }

  &__block-example>div {
    margin-bottom: 20px;
  }

  &__example+&__example,
  &__block-example+&__block-example {
    margin-top: map.get(variables.$spacing, 'size-5');
  }
}

$-checkbox-img: url("data:image/svg+xml,<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 20 20'><path fill='none' stroke='#fff' stroke-linecap='round' stroke-linejoin='round' stroke-width='3' d='M6 10l3 3l6-6'/></svg>") !default;

.page-main {
  position: relative;
  grid-area: main;
  padding-bottom: map.get(variables.$spacing, 'size-15');
  padding-top: map.get(variables.$spacing, 'size-15');

  @include bp('tablet-up') {
    padding-bottom: map.get(variables.$spacing, 'size-30');
    padding-top: map.get(variables.$spacing, 'size-30');
  }

  &__toolbar {
    display: flex;
    position: absolute;
    right: 2rem;
    top: 2.25rem;
    justify-content: flex-end;
  }

  &__checkbox {
    appearance: none;
    background-color: var(--pkt-color-input-background-normal);
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
    border: 2px solid var(--pkt-color-input-border-normal);
    display: inline-block;
    margin: 0.125rem 0.25rem 0 0;
    width: 1.25rem;
    height: 1.25rem;
    transition: all 0.3s ease-out;
    vertical-align: top;

    &[type='checkbox'] {
      border-radius: 0;
    }

    &:checked {
      &[type='checkbox'] {
        background-color: var(--pkt-color-input-border-normal);
        background-image: functions.escape-svg($-checkbox-img);
        border-color: var(--pkt-color-input-border-normal);
      }
    }
  }
}
