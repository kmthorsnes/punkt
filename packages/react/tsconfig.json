{
  "compilerOptions": {
    "target": "ESNext",
    "useDefineForClassFields": true,
    "lib": [
      "DOM",
      "DOM.Iterable",
      "ESNext"
    ],
    "allowJs": false,
    "skipLibCheck": true,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx",
    "baseUrl": ".",
    "paths": {
      "@/*": [
        "src/*"
      ],
      "pkt/*": [
        "node_modules/@oslokommune/punkt-css/dist/*"
      ],
      "pktAssets/*": [
        "node_modules/@oslokommune/punkt-assets/dist/*"
      ],
      "componentSpecs/*": [
        "../../component-specs/*"
      ],
      "punkt-react": [
        "src/components/index.ts"
      ]
    },
  },
  "include": [
    "src"
  ],
  "references": [
    {
      "path": "./tsconfig.node.json"
    }
  ]
}