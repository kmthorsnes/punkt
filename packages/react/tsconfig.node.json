{
  "compilerOptions": {
    "composite": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "types": [
      "jest",
      "node"
    ]
  },
  "include": [
    "vite.config.ts",
    "package.json"
  ]
}