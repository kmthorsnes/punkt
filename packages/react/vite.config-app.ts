import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import { resolve } from 'path'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      "@": resolve(__dirname, "src"),
      "pkt": "@oslokommune/punkt-css/dist",
      "pktAssets": "@oslokommune/punkt-assets/dist",
      "componentSpecs": resolve(__dirname, "../../component-specs/"),
    },
  },
  build: {
    outDir: "./dist-app",
  },
});
