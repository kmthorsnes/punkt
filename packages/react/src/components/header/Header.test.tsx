import '@testing-library/jest-dom/extend-expect'

import { fireEvent, render, screen } from '@testing-library/react'
import { axe, toHaveNoViolations } from 'jest-axe'
import React from 'react'

import { PktHeader } from './Header'

expect.extend(toHaveNoViolations)

describe('PktHeader', () => {
  const mockUser = {
    name: 'John Doe',
    shortname: 'JD',
    lastLoggedIn: '2023-08-20T12:34:56Z',
  }

  const mockUserMenu = [
    { title: 'Profile', target: '/profile', iconName: 'profile' },
    { title: 'Settings', target: '/settings', iconName: 'settings' },
  ]

  const mockRepresenting = {
    name: 'Org Name',
    shortname: 'ON',
    orgNumber: '123456789',
  }

  const mockUserMenuFooter = [{ title: 'Help', target: '/help' }]

  it('renders header with default props', () => {
    render(<PktHeader />)
    // Add assertions to check the presence of certain elements or classes
    expect(screen.getByTestId('pkt-header')).toBeInTheDocument()
  })

  it('renders user menu when user is present', () => {
    render(<PktHeader user={mockUser} userMenu={mockUserMenu} />)
    const userButton = screen.getByRole('button', { name: 'John Doe JD' })
    fireEvent.click(userButton)
    // Add assertions to check if user menu items are displayed
    expect(screen.getByText('Profile')).toBeInTheDocument()
    expect(screen.getByText('Settings')).toBeInTheDocument()
  })

  it('renders representing organization when representing is present', () => {
    render(<PktHeader representing={mockRepresenting} />)
    // Add assertions to check if representing organization info is displayed
    expect(screen.getAllByText('Org Name')[1]).toBeInTheDocument()
    expect(screen.getByText('Org.nr. 123456789')).toBeInTheDocument()
  })

  it('calls logOut function when Log Out button is clicked', () => {
    const mockLogOut = jest.fn()
    render(<PktHeader showLogOutButton logOut={mockLogOut} />)
    const logOutButton = screen.getByRole('button', { name: 'Logg ut' })
    fireEvent.click(logOutButton)
    expect(mockLogOut).toHaveBeenCalled()
  })

  it('toggles user menu when user button is clicked', () => {
    render(
      <PktHeader user={mockUser} userMenu={mockUserMenu} userMenuFooter={mockUserMenuFooter} />,
    )
    const userButton = screen.getByRole('button', { name: 'John Doe JD' })
    fireEvent.click(userButton)
    // Add assertions to check if user menu is open
    expect(screen.getByTestId('usermenu').classList.contains('pkt-header--open-dropdown')).toBe(
      true,
    )
    fireEvent.click(userButton)
    // Add assertions to check if user menu is closed
    expect(screen.getByTestId('usermenu').classList.contains('pkt-header--open-dropdown')).toBe(
      false,
    )
  })

  describe('accessibility', () => {
    it('renders with no wcag errors with axe', async () => {
      const { container } = render(
        <PktHeader user={mockUser} userMenu={mockUserMenu} userMenuFooter={mockUserMenuFooter} />,
      )
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })
  })

  describe('snapshot', () => {
    it('renders correctly with all props', () => {
      const { container } = render(
        <PktHeader
          user={mockUser}
          userMenu={mockUserMenu}
          userMenuFooter={mockUserMenuFooter}
          representing={mockRepresenting}
        />,
      )
      expect(container).toMatchSnapshot()
    })
  })
})
