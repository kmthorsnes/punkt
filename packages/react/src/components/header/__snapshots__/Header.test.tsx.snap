// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PktHeader snapshot renders correctly with all props 1`] = `
<div>
  <header
    aria-label="Topp"
    class="pkt-header pkt-header--fixed pkt-header--scroll-to-hide"
    data-testid="pkt-header"
    id="pkt-header"
  >
    <div
      class="pkt-header__logo"
    >
      <a
        aria-label="Tilbake til forside"
        class="pkt-header__logo-link"
        href="https://www.oslo.kommune.no/"
      />
      <span
        class="pkt-header__logo-service"
      />
    </div>
    <nav
      class="pkt-header__actions"
    >
      <ul
        class="pkt-header__actions-row"
      >
        <li
          class="pkt-header--has-dropdown "
          data-testid="usermenu"
        >
          <button
            aria-controls="pktUserDropdown"
            aria-expanded="false"
            class="pkt-header__user-btn pkt-btn pkt-btn--secondary pkt-btn--icons-right-and-left"
            role="button"
            type="button"
          >
            <span
              class="pkt-header__user-fullname"
            >
              Org Name
            </span>
            <span
              class="pkt-header__user-shortname"
            >
              ON
            </span>
          </button>
          <ul
            class="pkt-header__dropdown pkt-user-menu"
            id="pktUserDropdown"
          >
            <li>
              <div
                class="pkt-user-menu__label"
              >
                Pålogget som
              </div>
              <div
                class="pkt-user-menu__name"
              >
                John Doe
              </div>
              <div
                class="pkt-user-menu__last-logged-in"
              >
                Sist pålogget: 
                <time>
                  2023-08-20T12:34:56Z
                </time>
              </div>
            </li>
            <li>
              <ul
                class="pkt-list"
              >
                <li>
                  <a
                    class="pkt-link"
                    href="/profile"
                  >
                    Profile
                  </a>
                </li>
                <li>
                  <a
                    class="pkt-link"
                    href="/settings"
                  >
                    Settings
                  </a>
                </li>
              </ul>
            </li>
            <li>
              <div
                class="pkt-user-menu__label"
              >
                Representerer
              </div>
              <div
                class="pkt-user-menu__name"
              >
                Org Name
              </div>
              <div
                class="pkt-user-menu__org-number"
              >
                Org.nr. 
                123456789
              </div>
              <ul
                class="pkt-list mt-size-16"
              >
                <li>
                  <button
                    class="pkt-link-button pkt-link"
                  >
                    Endre organisasjon
                  </button>
                </li>
              </ul>
            </li>
            <li>
              <ul
                class="pkt-list"
              >
                <li>
                  <button
                    class="pkt-link-button pkt-link"
                  >
                    Logg ut
                  </button>
                </li>
              </ul>
            </li>
            <li
              class="footer"
            >
              <ul
                class="pkt-list-horizontal bordered"
              >
                <li>
                  <a
                    class="pkt-link"
                    href="/help"
                  >
                    Help
                  </a>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
